plane/source/docs/guided-mode.rst

```diff
@@ -4,10 +4,18 @@
 GUIDED Mode
 ===========
 
-The GUIDED mode is used when you want the aircraft to fly to a specific
-point on the map without setting up a mission. Most ground control
+GUIDED mode is used when you want the aircraft to fly to a specific
+point on the map, or in specific direction (heading) without setting up a mission. 
+
+Most ground control
 stations support a "click to fly to" feature where you can click a point
-on the map and the aircraft will fly to that location then loiter.
+on the map and the aircraft will fly to that location then loiter. If an 
+altitude is requested, the aircraft will fly at the current altitude until
```

Comment:
It will fly to the new altitude immediately. The point hast to be far enough away that it can finish the altitude change before it get there of course.  You could double check in SITL.

Reply:
I've done extensive testing in SITL - this is what I'm seeing.

Reply:
Also see ArduPlane/navigation.cpp:45 where the comment (and code) specifically says:

```
 // once we reach the position target we start checking the
 // altitude target
```

Reply:
I'm sure. To check I have also run in SITL. Very easy, mode takeoff arm (takes off to 50m), guided fly to somewhere a long way away at 100m. We get this:
![image](https://github.com/ArduPilot/ardupilot_wiki/assets/33176108/c701007a-e4d6-42c3-90ce-30d7c7e27b19)

Climb starts immediately, 100m is reached well before destination.

Note that if its a climb it will climb as fast as possible, where as a decent will glide such that it reaches the destination and the altitude at the same time.

The only case where it will still be changing altitude once it has reached the destination is if the change in altitude is sufficiently large it cannot be achieved in the available horizontal distance. 

The code your looking at has nothing to do with the setting target altitudes, it is what checks when we have reached the desired location so we can move on to the next waypoint for example.

How have you been testing?

Reply:
I'm running two arduplane SITL binaries, one of them is sending MAVLink updates to the other, the 2nd is using AP_Follow to receive the target plane info and then Lua script to call the new Lua run_command_int DO_REPOSITION command.

Reply:
I think I see the problem I have @IamPete1 - my code calls DO_REPOSITION at 10Hz as it gets new position data from the plane it is following via MAVLink. I think this means it stays on the first part of your graph - where the altitude is level - and never gets a chance to head up before I hit it with a new position.